File: cpp/1-two_sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> tempMap;
        vector<int> result;
        for(int i=0;i<nums.size();i++){
            if(tempMap.find(target-nums[i])!=tempMap.end())
            {
                result.push_back(tempMap[target-nums[i]]);
                result.push_back(i);
            }
            tempMap[nums[i]] = i;
        }
        return result;
    }
};

File: python/1-two_sum.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        temp_dict = {}
        for i in range(len(nums)):
            if nums[i] in temp_dict:
                return [temp_dict[nums[i]],i]
            temp_dict[target-nums[i]] = i